Lecture3/Task1/Task1L3.java

```diff
@@ -0,0 +1,246 @@
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Task1L3 {
+
+    private static List<Voucher> listOfVouchers = new ArrayList<>();
+    private static List<Voucher> listOfSelection = new ArrayList<>();
+    private static Map<Integer, Voucher> mapOfSelection = new TreeMap<>();
+
+    public static void main(String[] args) {
+        createRandomVouchers();
+        int selection =  getUserSelection();
+        provideFirstOffer(listOfVouchers, selection);
+        String advanced = needAdvancedSearch();
+        if (advanced.equalsIgnoreCase("n")) {
+            selectVoucher();
+        }
+        if (advanced.equalsIgnoreCase("y")) {
+            searchDaysCount();
+            searchFood();
+            searchTransfer();
+            provideFinalOffer();
+        }
+    }
+
+    private static void createRandomVouchers() {
+        for (int i = 50; i != 0; i--) {
+            int creationType = (int)(Math.random()*3);
+            if (creationType == 0) {
+                RestVoucher restVoucher = new RestVoucher();
+                restVoucher.initVoucher(i);
+                listOfVouchers.add(restVoucher);
+            }
+            else if (creationType == 1){
+                CureVoucher cureVoucher = new CureVoucher();
+                cureVoucher.initVoucher(i);
+                listOfVouchers.add(cureVoucher);
+            }
+            else {
+                ShoppingVoucher shoppingVoucher = new ShoppingVoucher();
+                shoppingVoucher.initVoucher(i);
+                listOfVouchers.add(shoppingVoucher);
+            }
+        }
+    }
+
+    private static int getUserSelection() {
+        int selection = 0;
+        while (selection == 0) {
+            System.out.print("Lets select needed type of the vouchers (1-Rest, 2-Cure, 3-Shopping): ");
+            Scanner input = new Scanner ( System.in );
+            int check;
+            try {
+                check = input.nextInt();
+            }
+            catch (Exception e) {
+                continue;
+            }
+            if (check >= 1 && check <= 3) {
+                selection = check;
+            }
+        }
+        return selection;
+    }
+
+    private static void provideFirstOffer (List<Voucher> listOfVouchers, int selection) {
```

Comment:
listOfVouhers is a class variable. Why do you use it as param in method?

Reply:
fixed